WalletWasabi/Gma/QrCodeNet/Encoding/DataEncodation/DataEncode.cs

```diff
@@ -82,13 +85,16 @@ internal static EncodationStruct Encode(IEnumerable<byte> content, ErrorCorrecti
 			dataCodewords.TerminateBites(dataCodewords.Count, vcStruct.VersionDetail.NumDataBytes);
 
 			int dataCodewordsCount = dataCodewords.Count;
-			if ((dataCodewordsCount & 0x7) != 0)
+			if ((dataCodewordsCount & 0x7) == 0)
 			{
-				throw new ArgumentException("datacodewords is not byte sized.");
+				if (dataCodewordsCount >> 3 != vcStruct.VersionDetail.NumDataBytes)
```

Comment:
Statement can be combined with above

Reply:
No it can't be combined.
If the first statement is true and the second is false the else part will be reached which is wrong.